### ISSUE TITLE
FTW validation against MapBiomas Argentina — Pampa region

### ISSUE THREAD

**Issue by ramon-pa** [fieldsoftheworld/ftw-baselines#262]
# FTW validation against MapBiomas Argentina — Pampa region

## Context

I ran an independent validation of the FTW global product (EfficientNet-B7) over the Argentine Pampa, using MapBiomas Argentina Collection 2 as the reference land cover. Argentina is not in the FTW training set (only Brazil represents South America in the paper), so this is a clean out-of-distribution test.

## Method

For each FTW polygon, I sampled the dominant MapBiomas 2024 land cover class (mode reducer, 10 m) and assigned it to one of four buckets:

- **TP** — cropland: codes `[18, 19, 39, 41, 48, 20, 40, 62]`
- **FP-pasture** — implanted grassland: code `[15]`
- **FP-natural** — natural vegetation: codes `[3, 4, 9, 11, 12, 13]`
- **FP-other** — everything else

The FP-pasture and FP-natural categories correspond directly to the out-of-scope classes listed in the paper's Limitations section — so these are expected false positives, not model failures.

Analysis run in Google Earth Engine. FTW polygons downloaded from the Explorer (full dataset, no confidence pre-filter applied at download time). Confidence filter applied manually in GEE using the `conf_mean` property.

## Results

**Tile S35W061** — B7 global product, 1° × 1°, ~10,200 km²

| | Unfiltered (n = 16,930) | conf_mean ≥ 0.4 — paper default (n = 14,741) |
|---|---|---|
| TP — cropland | 12,873 (76.0%) | 11,324 (76.8%) |
| FP — pasture | 1,721 (10.2%) | 1,509 (10.2%) |
| FP — natural vegetation | 2,192 (12.9%) | 1,758 (11.9%) |
| FP — other | 145 (0.9%) | 150 (1.0%) |
| **Polygon precision** | **76.04%** | **76.82%** |

**Historical tile 20HNH** — B5 local CLI, ~400 km²

| | |
|---|---|
| Polygons (≥ 1 ha) | 213 |
| Polygon precision | 74.2% |
| Pixel F1 | 88.4% |

The interesting finding is the consistency. Despite an 80× difference in sample size and different model sizes (B5 vs B7), the error structure is nearly identical across both tiles:

| | 20HNH (B5) | S35W061 (B7) |
|---|---|---|
| Precision | 74.2% | 76.0% |
| FP-pasture | 10.3% | 10.2% |
| FP-natural | 12.7% | 12.9% |

That stability suggests the ~23% false positive rate in the Pampa is systematic and predictable — which makes it a good candidate for post-processing.

For reference, pixel F1 on 20HNH (0.883) compares well against the in-training LPIS countries reported in Table 2 of the paper (Austria 0.892, Latvia 0.870, Finland 0.745), despite Argentina not appearing in the training set.

## Proposed next step: MapBiomas post-processing filter for the CLI

Isaac suggested contributing a MapBiomas-based filter to the CLI. The idea is a post-processing step that drops FTW polygons whose dominant MapBiomas class falls outside the cropland codes — essentially a land cover mask applied after inference.

MapBiomas currently covers: Argentina, Bolivia, Brazil, Chile, Colombia, Ecuador, Indonesia, Paraguay, Peru, Uruguay, and Venezuela — most of South America plus Indonesia, so the filter would have meaningful geographic reach.

Happy to discuss the right interface. Should this be an optional flag on an existing command (`ftw polygonize`?), or a separate `ftw filter` subcommand? Open to whatever fits best with the current CLI structure.

## Request: GEE asset for FTW predictions

To scale this validation to more tiles without downloading each one individually from the Explorer, it would be very helpful to have the FTW prediction vectors (or the underlying softmax raster) published as a Google Earth Engine asset. Right now the workflow requires a manual download per 1° × 1° tile from fieldsofthe.world, a local conversion step, and a manual UI upload into GEE — roughly 10–15 minutes per tile. With ~12 tiles covering the Argentine Pampa alone, a GEE-native asset would make regional-scale validation much more practical. Is this something that could be made available?

<img width="1166" height="1081" alt="Image" src="https://github.com/user-attachments/assets/9932aac5-559f-4ca4-99d4-22fbc9e566fb" />

**Comment by m-mohr** [comment]
Just a quick note before the academics will look over it:
The download functionality is downloading the full dataset without applying the confidence filter. So you need to apply the > 75% confidence filter manually, I hope it's just an ambiguous textual representation in point 1 of your graphic.

**Comment by ramon-pa** [comment]
Thanks @m-mohr for the catch. The download does include all polygons unfiltered. I re-ran with conf_mean ≥ 0.4 (paper default) and updated the figure. Results barely change: 76.82% vs 76.04% unfiltered, same error structure. 

Do you know what property the Explorer's slider uses? conf_mean in this tile tops out well below 0.75.

**Comment by m-mohr** [comment]
The UI uses `confidence_mean`, but stretches the values from 0 - 0.578178 to 0 - 100.  
0.578178 is the maximum value of the `confidence_mean` values (if we didn't fuck up determining the maximum value).
So for your initial 75% you would have to check for values >= 0.4336335.
